File: 프로그래머스/2/42584. 주식가격/주식가격.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

vector<int> solution(vector<int> prices) 
{
    vector<int> result(prices.size());
    for(int i=0; i<prices.size() - 1; i++)
    {
        int cnt = 1;
        for(int j=i+1; j<prices.size() - 1; j++)
        {
            if(prices[i] <= prices[j])
            {
                cnt++;
            }
            else
            {
                break;
            }
        }
        result[i] = cnt;
    }
    result[result.size() - 1] = 0;
    // for(int r : result)
    // {
    //     cout << r << " ";
    // }
    return result;
}

File: 프로그래머스/2/42584. 주식가격/주식가격.cs
using System;

public class Solution 
{
    public int[] solution(int[] prices) 
    {
        int[] result = new int[prices.Length];
        for(int i=0; i<prices.Length-1; i++)
        {
            int start = prices[i];
            int count = 0;
            for(int j=i+1; j<prices.Length; j++)
            {
                count++;
                if(start > prices[j])
                {
                    break;
                }
            }
            result[i] = count;
        }
        return result;
    }
}